code/datums/elements/turf_transparency.dm

```diff
@@ -56,20 +53,20 @@
 	if(dir != DOWN)
 		return
 
-	update_multiz(our_turf)
+	update_multi_z(our_turf)
 
 /datum/element/turf_z_transparency/proc/on_multiz_turf_new(turf/our_turf, turf/below_turf, dir)
 	SIGNAL_HANDLER
 
 	if(dir != DOWN)
 		return
 
-	update_multiz(our_turf)
+	update_multi_z(our_turf)
 
 ///Called when there is no real turf below this turf
-/datum/element/turf_z_transparency/proc/show_bottom_level(turf/our_turf)
-	if(!show_bottom_level)
-		return FALSE
+/datum/element/turf_z_transparency/proc/add_baseturf_underlay(turf/our_turf)
+	if(isgroundlessturf(our_turf)) // we don't ever want our bottom turf to be openspace
+		our_turf.ChangeTurf(/turf/open/floor/plating, flags = CHANGETURF_INHERIT_AIR)
```

Comment:
I really am not a fan of this, and would prefer it to just be a var option on the element add, unless there's a strong reason to not do that.

Typechecks in component initializes is fine, we often do it so a component can support some general "idea", with unique implementations for different types of objects. Only inhand items can honk on use for instance.

I don't like this sort of pattern, because it's tying the behavior of the element to something that's not inherent to the types you're checking for, and may even play poorly with them.

Lava for instance may not want to be scraped to a plating, or want to receive the openspace plane. Going from this to a bespoke typecache for multiz transparency only also seems fraught, since then rather then changing the arguments of the element, we're forced to seek out the correct list of types and add ourselves to it. Makes things easier to fuck up.

I realize I'm being hardheaded about this, but it seems like the wrong way to go with this solution. I realize the clutter you were trying to solve, but while removing useless args is very good, I'm not really convinced this is in the same category.

Reply:
Alright I'll look into fixing this tonight

Reply:
Alright it's been updated with the proposed changes.